# Generated by Django 2.1.3 on 2019-01-12 15:45

from django.conf import settings
from django.db import migrations, models
import django.db.models.deletion


class Migration(migrations.Migration):

    dependencies = [
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
        ('polygon', '0013_auto_20181106_1212'),
    ]

    operations = [
        migrations.CreateModel(
            name='Package',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
            ],
        ),
        migrations.CreateModel(
            name='CodeforcesPackage',
            fields=[
                ('package_ptr', models.OneToOneField(auto_created=True, on_delete=django.db.models.deletion.CASCADE, parent_link=True, primary_key=True, serialize=False, to='polygon.Package')),
                ('dir_name', models.CharField(max_length=64)),
                ('remote_problem_id', models.CharField(max_length=64)),
                ('status', models.IntegerField(choices=[(-1, 'Pending'), (0, 'OK'), (1, 'Failed')], default=-1)),
                ('create_time', models.DateTimeField(auto_now_add=True)),
                ('running_time', models.FloatField(null=True)),
                ('short_name', models.CharField(blank=True, max_length=192, null=True)),
                ('revision', models.IntegerField(null=True)),
            ],
            bases=('polygon.package',),
        ),
        migrations.AddField(
            model_name='package',
            name='created_by',
            field=models.ForeignKey(null=True, on_delete=django.db.models.deletion.SET_NULL, to=settings.AUTH_USER_MODEL),
        ),
    ]
